mono/metadata/assembly-internals.h

```diff
@@ -5,8 +5,33 @@
 #ifndef __MONO_METADATA_ASSEMBLY_INTERNALS_H__
 #define __MONO_METADATA_ASSEMBLY_INTERNALS_H__
 
+#include <glib.h>
+
 #include <mono/metadata/assembly.h>
 
 MONO_API MonoImage*    mono_assembly_load_module_checked (MonoAssembly *assembly, uint32_t idx, MonoError *error);
 
+/* If predicate returns true assembly should be loaded, if false ignore it. */
+typedef mono_bool (*MonoAssemblyCandidatePredicate)(MonoAssembly *, gpointer);
+
+MonoAssembly*          mono_assembly_open_predicate (const char *filename,
+						     mono_bool refonly,
+						     MonoAssemblyCandidatePredicate pred,
+						     gpointer user_data,
+						     MonoImageOpenStatus *status);
+
+MonoAssembly*          mono_assembly_load_from_predicate (MonoImage *image, const char *fname,
+							  mono_bool refonly,
+							  MonoAssemblyCandidatePredicate pred,
+							  gpointer user_data,
+							  MonoImageOpenStatus *status);
+
+/* MonoAssemblyCandidatePredicate that compares the assembly name (name, version,
+ * culture, public key token) of the candidate with the wanted name, if the
+ * wanted name has a public key token (if not present, always return true).
+ * Pass the wanted MonoAssemblyName* as the user_data.
+ */
+gboolean
+mono_assembly_candidate_predicate_sn_same_name (MonoAssembly *candidate, gpointer wanted_name);
```

Comment:
Well this way I can say `mono_assembly_open_predicate (..., &mono_assembly_candidate_predicate_sn_same_name, ...)` without a cast.